### PR TITLE
add e2e cloudstack aws iam and oidc

### PR DESCRIPTION
**Pull request by ddjjia**
*Issue #, if available:*
[E2E tests#1237](https://github.com/aws/eks-anywhere-internal/issues/1237)

*Description of changes:*

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/executables/executables.go [hunk 1]
```diff
@@ -88,9 +88,9 @@ func (e *executable) Close(ctx context.Context) error {
 func RedactCreds(cmd string, envMap map[string]string) string {
 	redactedEnvs := []string{}
 	for _, redactedEnvKey := range redactedEnvKeys {
-		if env, found := os.LookupEnv(redactedEnvKey); found {
+		if env, found := os.LookupEnv(redactedEnvKey); found && env != "" {
```
Comment (ddjjia) [inline pkg/executables/executables.go:91]: I have var read as empty string, which would add mask before and after each char

File: test/e2e/cloudstack_test.go [hunk 2]
```diff
@@ -1858,3 +1858,205 @@ func TestCloudStackMulticlusterWorkloadClusterGitHubFluxAPI(t *testing.T) {
...
+			framework.WithClusterName(test.NewWorkloadClusterName()),
+			framework.WithEnvVar(features.FullLifecycleAPIEnvVar, "true"),
+			framework.WithAwsIamEnvVarCheck(),
+		).WithClusterConfig(
+			api.ClusterToConfigFiller(
+				api.WithManagementCluster(managementCluster.ClusterName),
+				api.WithStackedEtcdTopology(),
+			),
+			framework.WithAwsIamConfig(),
+			cloudstack.WithRedhat123(),
```
Comment (cxbrowne1207) [inline test/e2e/cloudstack_test.go:1942]: Do we want to add another workload cluster to test it with version 1.24 as well?

File: test/e2e/cloudstack_test.go [hunk 3]
```diff
@@ -1858,3 +1858,205 @@ func TestCloudStackMulticlusterWorkloadClusterGitHubFluxAPI(t *testing.T) {
...
+			framework.WithClusterName(test.NewWorkloadClusterName()),
+			framework.WithEnvVar(features.FullLifecycleAPIEnvVar, "true"),
+			framework.WithOIDCEnvVarCheck(),
+		).WithClusterConfig(
+			api.ClusterToConfigFiller(
+				api.WithManagementCluster(managementCluster.ClusterName),
+				api.WithStackedEtcdTopology(),
+			),
+			framework.WithOIDCConfig(),
+			cloudstack.WithRedhat123(),
```
Comment (cxbrowne1207) [inline test/e2e/cloudstack_test.go:2043]: Here as well

File: test/e2e/cloudstack_test.go [hunk 4]
```diff
@@ -1858,3 +1858,205 @@ func TestCloudStackMulticlusterWorkloadClusterGitHubFluxAPI(t *testing.T) {
...
+			framework.WithClusterName(test.NewWorkloadClusterName()),
+			framework.WithEnvVar(features.FullLifecycleAPIEnvVar, "true"),
+			framework.WithAwsIamEnvVarCheck(),
+		).WithClusterConfig(
+			api.ClusterToConfigFiller(
+				api.WithManagementCluster(managementCluster.ClusterName),
+				api.WithStackedEtcdTopology(),
+			),
+			framework.WithAwsIamConfig(),
+			cloudstack.WithRedhat123(),
```
Comment (cxbrowne1207) [inline test/e2e/cloudstack_test.go:1889]: Do we want to add another workload cluster to test it with version 1.24 as well?

Comment (ddjjia) [reply test/e2e/cloudstack_test.go:1948]: added kubectl in 1.24

**Review by cxbrowne1207 — APPROVED** [review]

**Pull request merged**